Fix the torch version parsing logic

houseroad opened the pull request:
Summary: In fbcode, importlib.metadata.version doesn't work (no PKG-INFO), so we need to switch to a compatible way to load versions.

Reviewed By: henryoier

Differential Revision: D72221498

ywang96 commented on vllm/utils.py:
```diff
@@ -2580,3 +2581,12 @@ def sha256(input) -> int:
                           byteorder="big")
+
+
+def is_torch_equal_or_newer(target: str):
```
Could you please add return type hint and a short docstring?

houseroad replied:
sure, will do

houseroad replied:
Added

ywang96 reviewed (approved):
Left a comment - PTAL

houseroad committed:
Fix the torch version parsing logic (#15857)

Summary:
Pull Request resolved: https://github.com/vllm-project/vllm/pull/15857

In fbcode, importlib.metadata.version doesn't work (no PKG-INFO), so we need to switch to a compatible way to load versions.

Reviewed By: henryoier

Differential Revision: D72221498

luccafong reviewed (approved):
looks good!
